sql/src/main/java/io/crate/planner/projection/builder/InputCreatingVisitor.java

```diff
@@ -47,15 +47,17 @@
             this.inputs = new HashMap<>(inputs.size());
             int i = 0;
             for (Symbol input : inputs) {
-                DataType valueType = input.valueType();
-                if (fromStep != null && fromStep == Aggregation.Step.PARTIAL){
-                    // TODO: once we have the datatype of partial aggs, we need to add it here
-                    Function function = (Function) input;
-                    if (function.info().type() == FunctionInfo.Type.AGGREGATE){
-                        valueType = DataTypes.UNDEFINED;
+                if (!input.symbolType().isValueSymbol()) {
```

Comment:
maybe add a note that value symbols aren't replaced with InputColumn because of the scalar function compilation